### PR TITLE
Controller tests setup

### PR DESCRIPTION
**Pull request by live-wire**
Currently, we don't have any controller tests. 
To add more functionality(more CRDs?) safely, we should have a framework for adding FlinkCluster lifecycle tests. 

- Use explicit envtest versions
- Test to check if JobManager StatefulSet exists on new FlinkCluster creation

### CODE REVIEW

File: controllers/flinkcluster/suite_test.go [hunk 1]
```diff
@@ -98,7 +100,11 @@ var _ = BeforeSuite(func() {
 
 var _ = AfterSuite(func() {
 	cancel()
+	ctx.Done()
 	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).ToNot(HaveOccurred())
+	timeout := 30 * time.Second
+	poll := 5 * time.Second
+	Eventually(func() error {
```
Comment (live-wire) [inline controllers/flinkcluster/suite_test.go:107]: kube-apiserver can take some time to be terminated when the test suite is being terminated.
[https://github.com/kubernetes-sigs/controller-runtime/issues/1571](Known issue)

File: Makefile [hunk 2]
```diff
@@ -53,11 +60,11 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: test
-test: manifests generate fmt vet tidy kustomize envtest ## Run tests.
```
Comment (regadas) [inline Makefile:56]: I would keep tidy here; more often then not we forget about it when do it a PR

File: Makefile [hunk 3]
```diff
@@ -53,11 +60,11 @@ vet: ## Run go vet against code.
...
-	rm -rf config/test && mkdir -p config/test/crd
-	$(KUSTOMIZE) build config/crd > config/test/crd/flinkoperator.k8s.io_flinkclusters.yaml
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
```
Comment (regadas) [inline Makefile:59]: replacing only KUBEBUILDER_ASSETS wasn't working? the way it was should give you the same outcome

Comment (live-wire) [reply Makefile:59]: It wasn't for some reason :|

Comment (live-wire) [reply Makefile]: I cleared local cache of `bin` and the old way seems to work as well 👍

**Review by regadas — APPROVED** [review]
ok this seems like a starting point 😄

**Pull request merged**